lib/bounces.js

```diff
@@ -49,6 +49,44 @@ module.exports = function (log, error) {
       }
     }
 
+    function getHeaderValue(headerName, message){
+      var value = ''
+      if (message.mail && message.mail.headers) {
+        message.mail.headers.some(function (header) {
+          if (header.name === headerName) {
+            value = header.value
+            return true
+          }
+
+          return false
+        })
+      }
+
+      return value
+    }
+
+    function logBounceEvent(message) {
+      var verifyCode = getHeaderValue('X-Verify-Code', message)
+      var uid = getHeaderValue('X-Uid', message)
+
+      if (uid && verifyCode) {
+        var fakeRequestObject = {
+          auth: {
+            credentials: {
+              uid: uid,
+              id: verifyCode
+            }
+          },
+          headers: message.headers
+        }
+
+        return log.activityEvent('account.confirmation.bounced', fakeRequestObject, {
```

Comment:
Will this work though? I Think you need stuff like this https://github.com/mozilla/fxa-auth-server/pull/1431/files#diff-995141b6841f29fa041aca2b683b294fR12 because this is running in its own process.